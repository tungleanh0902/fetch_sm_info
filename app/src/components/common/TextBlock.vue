<template>
  <p class="text-block">
    {{ content.trim() }}
  </p>
</template>

<script>
export default {
  name: `text-block`,
  props: {
    content: {
      type: String,
      default: "",
    },
  },
}
</script>

<style scoped>
.text-block {
  word-break: break-word;
  font-size: 14px;
  white-space: pre-wrap;
}

.text-block p {
  line-height: 22px;
  margin-bottom: 1.5rem;
}

.text-block p,
.text-block ul,
.text-block ol,
.text-block blockquote,
.text-block pre {
  max-width: 40rem;
}
</style>
