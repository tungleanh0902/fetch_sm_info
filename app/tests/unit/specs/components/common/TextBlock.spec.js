import { mount } from "@vue/test-utils"
import TextBlock from "common/TextBlock"

describe(`TextBlock`, () => {
  let wrapper

  beforeEach(() => {
    wrapper = mount(TextBlock, {
      propsData: {
        content: `Hello World`,
        author: `faboweb`,
      },
    })
  })

  it(`has the expected html structure`, () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
