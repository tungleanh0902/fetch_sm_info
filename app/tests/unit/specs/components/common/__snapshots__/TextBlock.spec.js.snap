// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextBlock has the expected html structure 1`] = `
<p
  author="faboweb"
  class="text-block"
>
  
  Hello World

</p>
`;
